Extension/src/common.ts

```diff
@@ -534,6 +534,29 @@ export function checkInstallLockFile(): Promise<boolean> {
     return checkFileExists(getInstallLockPath());
 }
 
+/** Get the platform that the installed binaries belong to.*/
+export async function getInstalledBinaryPlatform(): Promise<string | undefined> {
+    // the debugAdapters folder is utilized to identify the platform
+    const path_to_check: string = getExtensionFilePath("debugAdapters");
+    let installedPlatform: string = "";
+    if (checkDirectoryExists(path_to_check)) {
+        const files: string[] = await readDir(path_to_check);
+        installedPlatform = "win32";
+        if (files !== undefined) {
+            (files).forEach((element: string) => {
```

Comment:
it looks like you are relying on checking to see which `mono`  runtime we have installed to figure out the installed platform. Is that the best way to do this? Is the `uname` method not working? Also, once you have the installedPlatform set, you should exit instead of looping through all the files in the folder.

Reply:
Is the bug to do with the language service or with the debugger?

Reply:
This is not related to debugger.  `uname` works, but we're trying to determine which "offline vsix" was installed and if it matches the OS is was installed on.  We need to look at something in the extension's files to determine this.  I think another alternative would be to embed a simple text file in the extension as part of our build process.  We already inject lock.txt when we build offline vsix'es.  We could potentially add content to lock.txt (currently it's empty) and just read it to see if it matches the OS.  If lock.txt is empty (like for online vsix), we don't do the check.

Reply:
This should be a bug on both. If a user decides to download the wrong offline installer it wont execute on their machine. 

`uname` will determine the current machine but does not determine the offline vsix.

I am not a fan of this solution since its assuming the debugger will always use a mono runtime with specific extensions to determine the offline package type.

Reply:
You could just check the LLVM/clang-format or clang-format.exe or clang-format.darwin.

Reply:
A possible solution for the future is to add a dummy file in the VSIX folder that indicates the VSIXplatform's info. For now, a unified solution for win/mac/linux32/linux64 is to check this folder. We can replace this with the name of the dummy file later on.

Reply:
We wouldn't need to add a new dummy file.  We could just add contents to install.lock and you can read the file.  But since we don't support 32-bit linux anymore, we can just look at clang-format as was discussed earlier today..

Reply:
In the new approach, we check the LLVM/bin folder.
![image](https://user-images.githubusercontent.com/16143955/80335340-d9001a80-8808-11ea-9017-6a4954fc5803.png)